src/app/services/actions/query-action-creators.ts

```diff
@@ -20,11 +20,11 @@ export function runQuery(query: IQuery): Function {
   let authToken = '{token:https://graph.microsoft.com/}';
   let graphUrl = `https://proxy.apisandbox.msdn.microsoft.com/svc?url=${query.sampleURL}`;
   const respHeaders: any = {};
-  const authenticated = localStorage.getItem('authenticated');
+  const authenticatedUser = localStorage.getItem('authenticatedUser');
```

Comment:
We are querying local storage everytime we make a request and in other places. This is not efficient. We can reduce the number of times we are fetching data from the local storage by loading authenticatedUser into the store when the application loads.

You can do this by passing `authenticatedUser` as the initial state when creating the store in `index.tsx`

Reply:
I used learnings from this article.  Local storage actually assists us and is cleared after use

https://hackernoon.com/how-to-take-advantage-of-local-storage-in-your-react-projects-a895f2b2d3f2

Reply:
What do you mean it is cleared after use? 

We are making calls to local storage in multiple places. We can reduce this by loading the data from local storage to the store when the page loads. This can be a maintanance nightmare especially when we decided to change the key with which we are storing the values.

It's also a perfomance overhead.